staging/src/k8s.io/apiserver/pkg/cel/types.go

```diff
@@ -429,7 +429,7 @@ func (rt *DeclTypeProvider) FindStructType(typeName string) (*types.Type, bool)
 	declType, found := rt.findDeclType(typeName)
 	if found {
 		expT := declType.CelType()
-		return expT, found
+		return types.NewTypeTypeWithParam(expT), found
```

Comment:
The returned celType is actually a typeType defined in https://github.com/google/cel-go/blob/7c131686c5fc852e9972934effe40e7cc6be5331/cel/decls.go#L131-L132
I don't understand why we need a duplicated conversion here.

Reply:
@cici37 it's indeed a `types.Type` (a `Type` where the kind is a `StructKind`). It's not what CEL expects, it wants a `types.Type` where the kind is a `TypeKind` wrapping the `Type` with the `StructKind`.